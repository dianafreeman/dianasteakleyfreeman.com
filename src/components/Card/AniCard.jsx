import React, { useRef, useEffect, useState } from 'react';
import { animated } from 'react-spring';

const AniCard = ({ children, styles, ...restProps }) => {
  const Wrapper = animated(Card);
  return <Wrapper {...restProps} style={styles} />;
};
