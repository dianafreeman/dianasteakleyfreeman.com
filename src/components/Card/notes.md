# Card
Lets get specific;

## The problem:
The shadows on white cards show up behind all of teh cards.

They should be visible whenever a card is active, no matter the display type or order.

Option 1:
    - does changing to display-flex fix this?

Option 2:
    - can I create a full content wrapper that shows a rectantular shadow (svg generated?) on hover?

## ToDo:
-- rearrange this shadow box method
-- There are too many components here
