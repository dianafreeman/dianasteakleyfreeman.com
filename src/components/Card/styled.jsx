import styled from 'styled-components';
import { Link } from 'gatsby';
import { animated } from 'react-spring';
import theme from '../../config/theme';

export const CardStage = styled.div`
  position: relative;
`;
export const CardWrapper = animated(styled(Link)`
  min-height: 200px;
  min-width: 200px;
  display: flex;
  position: relative;
  flex-basis: 100%;
  margin: 0.5em 1em;
  text-align: left;
  @media screen and (min-width: ${theme.screen.sm}) {
    flex-basis: ${props => (props.listIsEven ? '50%' : '100%')};
  }
`);

export const CardContent = animated(styled.div`
  width: 100%;
  background-color: white;
  margin: 0.5em;
  z-index: 0;
`);

export const ShadowWrapDiv = styled(CardContent)`
  position: absolute;
  top: -0.5em;
  left: -0.5em;
  width: 100%;
  height: 100%;
  background-color: ${props => theme.color[props.colorKey]};
  z-index: 0;
`;

export const CardTitle = styled.h1`
  color: ${theme.color.dark};
  width: 100%;
  margin-left: 0.5em;
  position: absolute;
  bottom: 0px;
  white-space: wrap;
`;
