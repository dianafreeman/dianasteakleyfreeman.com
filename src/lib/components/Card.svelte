<script>

  export let title;
  
  export let target;
  export let imgSrc;
</script>

<a href={target} class="flex p-4 relative text-md focus:outline-white focus:outline-1 grayscale hover:grayscale-0 aspect-square bg-light-gray hover:bg-white focus:bg-white active:bg-white active:outline active:outline-white active:outline-offset-5 text-black justify-end flex-col">
  <!-- <img src={imgSrc} alt="" class="w-full self-start object-contain" /> -->
  <!-- <div class="bottom-0 absolute w-full p-3" style="background-color: rgba(0,0,0,0.9);"> -->
  <h3 class="text-xl font-bold lowercase">{title}.</h3>
<!-- </div> -->

</a>
