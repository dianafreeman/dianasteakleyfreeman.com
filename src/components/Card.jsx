/* eslint-disable */
import React from "react";
import PropTypes from "prop-types";
import Img from 'gatsby-image'
import { Link } from 'gatsby'
import styled from "styled-components";
import tw from "tailwind.macro";
import { colors } from '../../tailwind';
import { useSpring, animated } from 'react-spring';

const CardWrapper = styled.div`
  ${tw`shadow-lg no-underline rounded-lg text-white relative text-center`};
  background: ${props => props.bg}98;
  min-height: 200px;
  min-width: 200px;
  margin: 1em;
`

const TextWrapper = styled.div`
  ${tw`font-sans text-small `};
  text-shadow: 0 2px 10px rgba(0, 0, 0, 0.3);
`;
const CardTitle = styled.h3`
  ${tw`font-serif`}
  font-weight: 700;

`;
const CardFooter = styled.div`
  ${tw`absolute w-full font-sans`}
  background-color: ${colors['grey-dark']};
  background: linear-gradient(to right,#474d51,#273238);
  bottom: 0;
`

const CardImage = styled(Img)`
  ${tw`absolute pin-t`}
  &:after{
    content: ' ';
    background-color: rgba(${props => props.overlay}, 0.9);
    position: absolute;
    top: 0;
    left: 0;
    right: 0;
    bottom: 0;
  }
`


const Card = ({ title, bg, media , slug, color, type}) => {
  return (
    <Link to={`/${type}/${slug}`}>
    <CardWrapper bg={bg}>
        <CardImage fluid={media.localFile.childImageSharp.fluid} overlay={color}/>
        <CardFooter>
        <CardTitle>{title}</CardTitle>
        </CardFooter>
    </CardWrapper>
    </Link>
  );
};
export default Card;

Card.propTypes = {
  title: PropTypes.string.isRequired,
  bg: PropTypes.string.isRequired
};
