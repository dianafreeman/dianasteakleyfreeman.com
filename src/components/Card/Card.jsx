import React, { useRef, useEffect, useState } from 'react';
import PropTypes from 'prop-types';
import { Link } from 'gatsby';
import styled from 'styled-components';
import tw from 'tailwind.macro';
import { useSpring, animated } from 'react-spring';

const CardWrapper = styled.div`
  ${tw`relative center inline-flex`};
  min-height: ${props => props.minHeight || '100' }px;
  margin: 10px;
  `;

const CardBody = styled.div`
  ${tw`w-1/2 relative mx-1 bg-card`}
  min-width: 33vw;
  `;

const CardContent = styled.div`
  ${tw`px-2 m-auto`}

`;
const CardTitle = styled.h3`
  ${tw`text-white w-full px-2 absolute pin-b mb-1`}
`


const Card = ({ title, onClick, children, ...restProps }) => {
  return (
    <CardWrapper {...restProps} {...restProps}>
      <CardBody>
        {children && <CardContent>{children}</CardContent>}
        <CardTitle>{title}</CardTitle>
      </CardBody>
    </CardWrapper>
  );
};
export default Card;

Card.propTypes = {
  title: PropTypes.string,
  onClick: PropTypes.func,
};
