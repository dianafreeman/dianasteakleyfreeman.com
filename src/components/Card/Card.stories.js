import React from 'react';
import { action } from '@storybook/addon-actions';
import { smallItem, withTheme } from '../../../.storybook/decorate';
import Card from '.';

const ITEM = {
  id: 2,
  title: 'Bio',
  destination: '/.bio',
};

export const byDefault = () => {
  return <Card item={ITEM} onClick={action('You clicked')} />;
};
export default {
  title: 'Card',
  component: Card,
  decorators: [smallItem, withTheme],
};
