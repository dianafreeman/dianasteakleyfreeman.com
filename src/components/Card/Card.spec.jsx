import React from 'react';
import renderer from 'react-testing-library';

import Card from '.';

describe('Card', () => {
  let component, props;

  beforeEach(() => {
    props = {};
    component = renderer.create(<Card {...props}> A Card Label</Card>);
  });

  it('should match snapshot', () => {
    expect(component).toMatchSnapshot();
  });
});
