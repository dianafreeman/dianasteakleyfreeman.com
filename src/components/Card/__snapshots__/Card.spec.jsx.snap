// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Card should match snapshot 1`] = `
.c0 {
  position: relative;
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  min-height: 100px;
  margin: 10px;
}

.c1 {
  width: 50%;
  position: relative;
  margin-left: 0.25rem;
  margin-right: 0.25rem;
  background-color: #B7B7B7;
  min-width: 33vw;
}

.c2 {
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  margin: auto;
}

.c3 {
  color: #ffffff;
  width: 100%;
  padding-left: 0.5rem;
  padding-right: 0.5rem;
  position: absolute;
  bottom: 0;
  margin-bottom: 0.25rem;
}

<div
  className="c0"
>
  <div
    className="c1"
  >
    <div
      className="c2"
    >
      blog
    </div>
    <h3
      className="c3"
    />
  </div>
</div>
`;
